jekyll/_cci2/install-machine-runner-3-on-docker.adoc

```diff
@@ -0,0 +1,110 @@
+:icons: font
+:experimental:
+:machine:
+:docker:
+
+This page describes how to install CircleCI's machine runner 3.0 with the Docker executor on server. If you are looking to set up self-hosted runners in a private Kubernetes cluster, visit the <<container-runner-installation#,Container runner>> page.
+
+[#new-recommended-method-container-runner]
+== Recommended method for Docker installation (container runner)
+
+The <<container-runner#,container runner>> is the recommended approach for running containerized jobs on self-hosted runners. Container runner offers the ability to seamlessly define, publish, and use custom Docker images during job execution, as well as the ability to manage dependencies or libraries through custom Docker images instead of enumerating dependencies as part of `steps` in the `.circleci/config.yml` file.
+
+[#machine-approach-with-docker]
+== Machine-based approach with Docker
+
+[#machine-runner-prerequsites]
+=== Prerequisites
+
+* link:https://docs.docker.com/engine/install/[Docker]
+
+[#additional-prerequisites]
+==== Additional prerequisites with Docker
+
+The host needs to have Docker installed. Once the `runner` container is started, the container will immediately attempt to start running jobs. The container will be reused to run more jobs indefinitely until it is stopped.
+
+The number of containers running in parallel on the host is constrained by the host's available resources and your jobs' performance requirements.
+
+[#self-hosted-runner-terms-agreement]
+=== Self-hosted runner terms agreement
+
+{% include snippets/runner/terms.adoc %}
+
+[#create-namespace-and-resource-class]
+=== 1. Create namespace and resource class
+
+{% include snippets/runner/install-with-cli-steps.adoc %}
+
+=== 2. Create a Dockerfile that extends the machine runner 3.0 image
+
+Create a `Dockerfile.runner.extended` file. In this example, Python 3 is installed on top of the base image.
+
+```dockerfile
+FROM circleci/runner-agent:machine-3
+RUN sudo apt-get update; \
+    sudo apt-get install --no-install-recommends -y \
+        python3
+```
+
+[#build-the-docker-image]
+=== 3. Build the Docker image
+
+```shell
+docker build --file ./Dockerfile.runner.extended .
+```
+
+[#start-the-docker-container]
+=== 4. Start the Docker container
+
+NOTE: The environment variable values are not available to the `docker` command, so these environment variables are not visible in `ps` output.
+
+```shell
+CIRCLECI_RUNNER_NAME=<runner-name> CIRCLECI_RUNNER_API_AUTH_TOKEN=<runner-token> docker run --env CIRCLECI_RUNNER_NAME --env CIRCLECI_RUNNER_API_AUTH_TOKEN --name <container-name> <image-id-from-previous-step>
+```
+
+When the container starts, it will immediately attempt to start running jobs.
+
+[#start-the-docker-container-on-server]
+==== Start the Docker container on server
+
+When starting the Docker container on server, the `agent_version` and `LAUNCH_AGENT_API_URL` environment variables will need to be passed in using the `--env` flag.
```

Comment:
s/LAUNCH_AGENT_API_URL/CIRCLECI_RUNNER_API_URL

Reply:
Maybe a hold over from the launch-agent docs? I don't think we specify an agent version any more, or at least I don't see it in the command options when calling help on the machine mode.